extension-ui/background.js

```diff
@@ -1,84 +1,127 @@
-
 let blockedSitesCache = null;
+let allowedSitesCache = ["https://chatgpt.com"];
+let isBlackList = false;
 
-// Initialize cache when the extension is loaded
-chrome.runtime.onStartup.addListener(() => initializeBlockedSitesCache());
-chrome.runtime.onInstalled.addListener(() => initializeBlockedSitesCache());
+function fetchUserData() {
+  fetch('http://localhost:5000/users/me', { 
```

Comment:
you should have a variable with baseUrl

Reply:
this is not baseUrl what you did... it's fullUrl.
you should have http://localhost:5000 in the base variable